inspectit-ocelot-agent/src/system-test/java/rocks/inspectit/ocelot/utils/TestUtils.java

```diff
@@ -76,20 +100,31 @@ public static void waitForClassInstrumentations(List<Class> clazzes, int duratio
             await().atMost(duration, timeUnit).ignoreExceptions().untilAsserted(() -> {
                 for (Class clazz : clazzes) {
                     sink = Class.forName(clazz.getName(), true, clazz.getClassLoader());
-                    Object clazzInstrumentation = getInstrumentationCache().getIfPresent(clazz);
-                    assertThat(clazzInstrumentation).isNotNull();
+                    Long timeStamp = instrumentationTimeStamp.get(clazz);
```

Comment:
I think this has to be reverted and we have to check the actual cache here. Because otherwise, assuming a class' instrumentation has been reverted, we would still consider it as instrumented because it exists in our local cache.
Maybe it's quite unlikly but I think there is no benefit checking the local cache (the one in this class)